### PR TITLE
Allow add-ons to customize the robots.txt

### PR DESCRIPTION
**Pull request by tburry**
This change makes a `Robots` data model that can be modified by add-ons to add sitemaps and add rules to the robots.txt file. As we add more applications we need a way for them to add their sitemaps to the application and this is meant to accomplish that.

### CODE REVIEW

**Comment by tburry** [comment]
As an FYI, this PR is a result of me fleshing out the requirements for KB sitemaps. Probably doesn't need to be approved this sprint if it's too much.

File: plugins/Sitemaps/Robots.php [hunk 1]
```diff
@@ -0,0 +1,70 @@
...
+
+    /**
+     * Get the list of robots rules.
+     *
+     * @return array Returns an array of rules.
+     */
+    public function getRules(): array {
+        return $this->rules;
+    }
+}
```
Comment (charrondev) [inline plugins/Sitemaps/Robots.php]: Missing a line ending here.

File: plugins/Sitemaps/class.sitemaps.plugin.php [hunk 2]
```diff
@@ -110,7 +110,7 @@ public function utilityController_robots_create($sender) {
         $sender->deliveryType(DELIVERY_TYPE_VIEW);
         $sender->setHeader('Content-Type', 'text/plain');
 
-        $robots = new \Vanilla\Sitemaps\Robots();
+        $robots = new \Vanilla\Sitemap\Robots();
```
Comment (tburry) [inline plugins/Sitemaps/class.sitemaps.plugin.php]: The namespace really should match the name of the addon which is plural.

**Comment by tburry** [comment]
WRT moving the functionality into core I agree with the sentiment, but I think that's too much yak shaving for this sentiment. Keep in mind that kb doesn't really depend on sitemaps to work. This would be more like a "suggested addon".

File: plugins/Sitemaps/class.sitemaps.plugin.php (hunk 2 above)
Comment (charrondev) [reply plugins/Sitemaps/class.sitemaps.plugin.php]: In any event, we've moved this class up into core now. I think the class is now `Vanilla\Web\Robots`.

**Pull request merged**